src/store/sagas/dataset-saga.js

```diff
@@ -1,11 +1,16 @@
 import { takeEvery } from 'redux-saga/effects'
-import { RETRIEVE_DATASET, datasetDetails } from '../actions'
+import { RETRIEVE_DATASET_REFERENCE, RETRIEVE_DATASET, datasetDetails, datasetReference } from '../actions'
 import apiInvoker from './api-invoker'
 
 export default function * theRealDatasetSaga () {
-  yield takeEvery(RETRIEVE_DATASET, invokeApiWithParameter)
+  yield takeEvery(RETRIEVE_DATASET, invokeApiWithNames)
+  yield takeEvery(RETRIEVE_DATASET_REFERENCE, invokeApiWithId)
 }
 
-const invokeApiWithParameter = ({ value }) => {
+const invokeApiWithNames = ({ value }) => {
```

Comment:
Cool. Someday ™️  we'll write a real service/repo pattern wrapper to abstract this stuff better